### PR TITLE
Add support for removal comment annotations

### LINKED ISSUE #99: PropTypes not removed for ES Class Inheritance
**./components/Parent.jsx**
```jsx
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

export default class Parent extends PureComponent {
    static propTypes = {
        parentProp: PropTypes.string.isRequired,
    }
}
```

**./components/Child.jsx**
```jsx
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import Parent from './Parent';

export default class Child extends Parent {
    static propTypes = {
        childProp: PropTypes.string.isRequired,
    }
}
```

### PR DESCRIPTION
**Pull request by EnoahNetzach**
Fixes #99 and adds support to HOCs.

### CODE REVIEW

File: test/fixtures/variable-comment-annotation/options.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "removeImport": true
```
Comment (oliviertassinari) [inline test/fixtures/variable-comment-annotation/options.json]: I don't think that we need that option. Sounds unrelated to the PR.

File: src/isAnnotatedForRemoval.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// @flow weak
+
+export default function (node) {
+  const comments = node.trailingComments || [];
+
+  return !!comments.find(({ value }) => value.trim() === 'remove-proptypes');
```
Comment (oliviertassinari) [inline src/isAnnotatedForRemoval.js]: Is `!!` needed here?

File: src/isAnnotatedForRemoval.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// @flow weak
+
+export default function (node) {
+  const comments = node.trailingComments || [];
```
Comment (oliviertassinari) [inline src/isAnnotatedForRemoval.js]: A leading comment would be an [alternative](http://astexplorer.net/#/gist/85f95a9dcb9f7ebf5577b163fcec99c2/415c7c4718e4d18479c2e905d912e1bd71771fa0):
```js
Foo1.propTypes = { // remove-proptypes
  bar1: PropTypes.string.isRequired,
}


class Foo2 extends React.Component {
  static propTypes = { // remove-proptypes
    bar2: PropTypes.string,
  };

  render() {}
}

```

File: test/fixtures/class-comment-annotation/actual.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import ParentComponent from './Parent'
+
+export default class Greeting extends ParentComponent {
```
Comment (oliviertassinari) [inline test/fixtures/class-comment-annotation/actual.js]: What about simplifying the **tests**? For instance, I like how we can better reduce noise with:
```js
class Foo1 extends ParentComponent {
  render() {}
}

Foo1.propTypes /* remove-proptypes */ = {
  bar1: PropTypes.string,
};
```

File: test/fixtures/class-comment-annotation/options.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "removeImport": true
```
Comment (oliviertassinari) [inline test/fixtures/class-comment-annotation/options.json]: I don't think that it's related to this test.

File: test/fixtures/variable-comment-annotation/options.json (hunk 1 above)
Comment (EnoahNetzach) [inline test/fixtures/variable-comment-annotation/options.json]: it was just there from where I copied the fixture

File: src/isAnnotatedForRemoval.js (hunk 3 above)
Comment (EnoahNetzach) [inline src/isAnnotatedForRemoval.js]: of course!

File: src/isAnnotatedForRemoval.js (hunk 2 above)
Comment (EnoahNetzach) [inline src/isAnnotatedForRemoval.js]: Not strictly necessary, just habit, so it returns a boolean.

File: test/fixtures/class-comment-annotation/actual.js (hunk 4 above)
Comment (EnoahNetzach) [inline test/fixtures/class-comment-annotation/actual.js]: again, I copied it from another fixture, no problems

File: test/fixtures/class-comment-annotation/options.json (hunk 5 above)
Comment (EnoahNetzach) [inline test/fixtures/class-comment-annotation/options.json]: as above

File: src/isAnnotatedForRemoval.js (hunk 2 above)
Comment (EnoahNetzach) [inline src/isAnnotatedForRemoval.js]: Should I keep it, or remove it?

Comment (oliviertassinari) [inline src/isAnnotatedForRemoval.js]: Let's keep it, I couldn't find a better function: https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/find.

File: src/isAnnotatedForRemoval.js (hunk 3 above)
Comment (oliviertassinari) [inline src/isAnnotatedForRemoval.js]: I feel like this one is less intrusive than the other one. I would rather only support on for simplicity and consistency of the codebase. What do you think?

Comment (EnoahNetzach) [inline src/isAnnotatedForRemoval.js]: I feel better with `trailingComments` (a la [flow](https://flow.org/blog/2015/02/20/Flow-Comments/)).

A minor flaw with `leadingComments` could occur when the declaration is [inlined](http://astexplorer.net/#/gist/593c06903a408bfff33d09db9e38144b/e2a7a6c559538427bba6e5ed025b9a7125923c16).
And I'm not sure how to search for comments this way, as opposed to leading the LHS.

Comment (oliviertassinari) [inline src/isAnnotatedForRemoval.js]: Fine, I won't use it anyway, I would rather have the users decide of the syntax 👍 .

File: src/isAnnotatedForRemoval.js (hunk 2 above)
Comment (EnoahNetzach) [inline src/isAnnotatedForRemoval.js]: > I couldn't **find** a better function

😛

**Comment by oliviertassinari** [comment]
@EnoahNetzach Thanks!

**Pull request merged**